### DISCUSSION TITLE
FE 内存逐步升高，排查发现一台be的publishVersionTask累积在FE未释放，请问可能是什么原因呢？

### DISCUSSION THREAD

**Discussion by user-GitHub-user** [apache/doris#28333] [A - General / Q&A] [answered]
doris version: 1.2.6

业务使用flink 实时同步数据到doris中
堆栈信息：
![8b2030a82e388691f8b9d265e6a3bd4e](https://github.com/apache/doris/assets/18843801/1569e9d9-647a-4ac9-9dae-5c996731c7db)

通过堆栈发现AgentTaskQueue 的 tasks 中有大量publishVersionTask,
使用 show proc '/tasks/PUBLISH_VERSION' 查看，发现只有一台be有几万个publish version task, 其余be为0，fe日志显示每次tabletReport 该 be 都有 republish transactions，同时 show proc '/transactions'; 查看transactions数量，运行中的transactions只有十几个，已完成还未移除的也只有几千，什么情况会导致transactions完成，而publish version task未移除呢？
fe log:
[fe.log](https://github.com/apache/doris/files/13657450/fe.log)

**Comment by LemonLiTree** [comment]
Test environment?

**Reply by user-GitHub-user** [reply]
in pro environment

**Reply by LemonLiTree** [reply]
It is recommended to upgrade to 2.0 and verify it in the test environment first.

**Comment by user-GitHub-user** [comment, marked as answer]
be日志有大量 failed to register task 打印出来，应该是1.2.6的be在某种情况下会死锁，导致FE republish version task收不到返回，一直未移除，我们通过重启有问题的be后，FE内存恢复正常，可能有偶发性，后续再观察看
